apm-agent-plugins/apm-apache-httpclient-plugin/src/main/java/co/elastic/apm/agent/httpclient/HttpAsyncRequestProducerWrapper.java

```diff
@@ -118,14 +115,14 @@ public void resetRequest() throws IOException {
 
     @Override
     public void close() throws IOException {
-        helper.recycle(this);
```

Comment:
the error was we recycled (which sets the delegate to null) before closing the delegate, right? why did this only fail under load?

Reply:
Not really load, the request producer may not have been closed in the test, or closed on a separate thread where the test is not affected 🤷‍♂